### PR TITLE
added profile pages to users

### PR DESCRIPTION
**Pull request by jpboliv**

### CODE REVIEW

File: app/controllers/profiles_controller.rb [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        if(@user.update(profile_params))
+            flash[:success] = 'Your profile has been updated!'
+            redirect_to profile_path(@user.user_name)
+        else
+            flash[:danger] = @user.errors.full_messages
+            render :edit
+        end
+    end
+
+    private
```
Comment (JFilipeG) [inline app/controllers/profiles_controller.rb]: NL

File: app/controllers/profiles_controller.rb [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    end
+
+    def profile_params
+        params.require(:user).permit(:avatar,:bio)
+    end
+
+    #only the user can edit his profile
+    def can_edit_profile
+        unless current_user == @user
+            flash[:danger] = "YOU SHOULDN'T BE HERE BOY!"
```
Comment (JFilipeG) [inline app/controllers/profiles_controller.rb]: Don't show errors like this, show a normal error message on an error that could actually appear

File: app/helpers/profiles_helper.rb [hunk 3]
```diff
@@ -0,0 +1,2 @@
+module ProfilesHelper
```
Comment (JFilipeG) [inline app/helpers/profiles_helper.rb]: This is why i prefer to create my own models by myself :p

File: app/views/posts/_post.html.erb [hunk 4]
```diff
@@ -3,7 +3,8 @@
     <div class="post-head">
       <div class="thumb-img"></div>
       <div class="user-name">
-        <%= post.user.user_name %>
+        <%#= post.user.user_name %>
```
Comment (JFilipeG) [inline app/views/posts/_post.html.erb]: just delete this

File: app/helpers/application_helper.rb [hunk 5]
```diff
@@ -18,4 +18,12 @@ def has_image(post)
         end
     end
 
+    def profile_avatar_select(user)
```
Comment (JFilipeG) [inline app/helpers/application_helper.rb]: I think you could just have a default value on a user, like adding this pic on an before_create on the user's model, then you just need to show the user's pic everywhere. Beware that this means you need to add a pic everytime you try to delete the current pic. Unless there's no way to delete the profile's pic :p

Comment (JFilipeG) [reply app/helpers/application_helper.rb]: Or if you want to do something like this, keep the logic on the model. if there's no image attatched to the user, show the default.

File: config/routes.rb [hunk 6]
```diff
@@ -9,4 +9,7 @@
     resources :comments
   end
 
+  get ':user_name', to: 'profiles#show', as: :profile
```
Comment (JFilipeG) [inline config/routes.rb]: You can do this in a more standard way....

Comment (jpboliv) [reply config/routes.rb]: This is so that I can use the path /user_name to get into a profile, instead of /user/user_name
Is there any way better?

Comment (JFilipeG) [reply config/routes.rb]: Why `user_name`? Doesn't even makes sense :p

Comment (jpboliv) [reply config/routes.rb]: To get something like www.instagram.com/@jpboliv - localhost.com:3000/jpboliv

Comment (JFilipeG) [reply config/routes.rb]: oh, sorry I failed to see the `:`, good work 👍

File: .gitignore [hunk 7]
```diff
@@ -21,3 +21,6 @@
 /yarn-error.log
 
 .byebug_history
+
+app/assets/images/
+public/system/
```
[the remark this answers is not in the document]
Comment (JFilipeG) [reply .gitignore]: Add new line here

File: app/controllers/profiles_controller.rb (hunk 1 above)
Comment (JFilipeG) [reply app/controllers/profiles_controller.rb]: NL here

File: app/helpers/application_helper.rb [hunk 8]
```diff
@@ -17,5 +17,13 @@ def has_image(post)
             return image_tag 'placeholder.jpg', id: 'image-preview', class: 'img-responsive'
         end
     end
+    
+    # def profile_avatar_select(user)
```
Comment (JFilipeG) [inline app/helpers/application_helper.rb]: Delete unnecessary comments

File: app/models/user.rb [hunk 9]
```diff
@@ -1,9 +1,20 @@
...
+    
+    has_many :posts, dependent: :destroy
+
+    has_attached_file :avatar, styles: { medium: "300x300>", thumb: "100x100>" }, default_url: "/images/:style/missing.png"
+    validates_attachment_content_type :avatar, content_type: /\Aimage\/.*\z/
+    
+    acts_as_voter
+
+    def current_avatar()
+        return avatar.url(:medium) if avatar.exists?
```
Comment (JFilipeG) [inline app/models/user.rb]: Nice code, well done!!!!!! I'm almost crying! oh... wait... 🤔

**Pull request merged**